tests/test_monitoring.py

```diff
@@ -293,13 +294,29 @@ def test_read_monitoring_data_not_configured(motion_controller, monitoring):
 @pytest.mark.eoe
 @pytest.mark.usefixtures("mon_set_freq")
 @pytest.mark.usefixtures("mon_map_registers")
+@pytest.mark.develop
 def test_read_monitoring_data_disabled(monitoring):
-    # TODO Add exception in function for this test case
     monitoring.configure_sample_time(0.8, 0)
-    test_output = monitoring.read_monitoring_data()
-    assert len(test_output[0]) == 0
+    with pytest.raises(IMMonitoringError):
+        monitoring.read_monitoring_data()
```

Comment:
I think that here you're testing if there is no data to read rather than if the monitoring is disabled.